apps/prairielearn/src/lib/preact-html.ts

```diff
@@ -0,0 +1,22 @@
+import { render } from 'preact-render-to-string/jsx';
+
+import { type HtmlSafeString, unsafeHtml } from '@prairielearn/html';
+import type { VNode } from '@prairielearn/preact-cjs';
+
+import { DEV_MODE } from './config.js';
+
+// This function must live outside of `preact.tsx` so that it can be imported
+// on the client. `preact.tsx` imports `@prairielearn/compiled-assets`, which
+// cannot be bundled for the browser.
+
+/**
+ * This function should only be called as a aid for migration from client-side code to Preact.
+ * It allows authoring a piece of client-side code as Preact.
+ * Server-side rendering (e.g. pages/foo.html.tsx) should use `renderHtml` instead.
```

Comment:
I don't understand this note. AFAICT, `renderHtml` is just directly wrapping this function?

Reply:
Done in [269a3c1](https://github.com/PrairieLearn/PrairieLearn/pull/12158/commits/269a3c18d2984f45a285b9092ce7927276537fab).